src/main/java/com/trainer/name/mapper/TrainerMapper.java

```diff
@@ -26,7 +27,7 @@ public interface TrainerMapper {
 
     @Insert("INSERT INTO trainers (name, email) VALUES (#{name}, #{email})")
     @Options(useGeneratedKeys = true, keyProperty = "id")
-    void insert(Trainer trainer);
+    void insert(Trainer trainer) throws DuplicateKeyException;
```

Comment:
```suggestion
    void insert(Trainer trainer);
```
でよいかと。
難しいのですが、DuplicateKeyExceptionは非検査例外（実行時例外とも言う）を継承しています。
DuplicateKeyException の javadoc をみると RuntimeException を継承しているのがわかります。
https://spring.pleiades.io/spring-framework/docs/current/javadoc-api/org/springframework/dao/DuplicateKeyException.html
<img width="625" alt="スクリーンショット 2024-01-30 22 04 47" src="https://github.com/Ito-0131/10th-Assignment/assets/62045457/fa189013-71ef-4bdc-8efc-be5143fbbbe4">

非検査例外は RuntimeException という言葉でJavaでは定義されています。
「非」と書いてあるからには検査例外もあります。

この 2 つの違いを調べるとそもそも検査例外は不要だなんていう論にも出会うかと思います。
今は理解できなくていいので少し深ぼってみるとよいです 😄 

非検査例外の使い方としてメソッドに throws をつける必要がないという特徴があります。
正確にはあってもなくてもいいです。
人まずはそういうもんだという理解でOKなので消しておきましょう。
こんな感じです。
```suggestion
    void insert(Trainer trainer);
```

Reply:
388e156abbe609b8857aa15a11a977d75fa9f4b6
throws DuplicateKeyExceptionを削除しました
![image](https://github.com/Ito-0131/10th-Assignment/assets/138368847/0ef693e9-4b79-4d58-8c1a-59042a9d97a2)

Reply:
DuplicateKeyExceptionがRuntimeExceptionのサブクラスであるため、メソッドの宣言にthrows句を追加する必要がないことはなんとなく頭に入りましたので、今後DuplicateKeyExceptionを使った際にはthrowsを付けないように気をつけます

Reply:
DuplicateKeyExceptionがRuntimeExceptionのサブクラスであるため、メソッドの宣言にthrows句を追加する必要がないことはなんとなく頭に入りましたので、今後DuplicateKeyExceptionを使った際にはthrowsを付けないように気をつけます